core/src/test/java/org/frankframework/validation/XSDTest.java

```diff
@@ -70,7 +76,11 @@ public void testAddNamespacesToSchema(String schemaLocation) throws Exception {
 		XSD xsd = getXSD(schemaLocation);
 		xsd.setAddNamespaceToSchema(true);
 
-		String actual = xsd.addTargetNamespace();
+		StringWriter writer = new StringWriter();
+		XMLStreamWriter w = XmlUtils.REPAIR_NAMESPACES_OUTPUT_FACTORY.createXMLStreamWriter(writer);
+		SchemaUtils.mergeXsdsGroupedByNamespaceToSchemasWithoutIncludes(mock(), Map.of(xsd.getNamespace(), Set.of(xsd)), w);
+		w.flush();
+		String actual = writer.toString();
```

Comment:
I removed the test-only method and replaced it in this test with the production-code the test invoked, however the production code has slightly different output because it strips schema-locations and the test-only method didn't.

Other than that this code does the same as the test-only method.